### PR TITLE
New Bitwise Ops IA (cheatsheet)

### PR DESCRIPTION
**Pull request by Sayan98**
## Background

- [X] New Instant Answer
  - [X] Cheat Sheets: **`Bitwise Operations Cheat Sheet`**

---

Instant Answer Page: https://duck.co/ia/view/bitwise_ops_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by Sayan98** [comment]
@amarlearning Hey, is anything wrong with the PR?

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=bitwise ops cheatsheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/bitwise_ops_cheat_sheet_mobile.png?QJYexLaVyJ)

File: share/goodie/cheat_sheets/json/bitwise-ops.json [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            {
+                "key": "<<",
+                "val": "The SHIFT RIGHT operator moves the bits to the right, discards the far right bit, and assigns the leftmost bit a value of 0. Each move to the right effectively divides op1 in half."
+            },
+            {
+                "key": ">>",
+                "val": "The SHIFT LEFT operator moves the bits to the left, discards the far left bit, and assigns the rightmost bit a value of 0. Each move to the left effectively multiplies op1 by 2."
+            }
+        ]
+    }
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/bitwise-ops.json]: You should also include `Zero-fill right shift` operator as well!

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=bitwise ops cheatsheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/bitwise_ops_cheat_sheet_mobile.png?d2K2m7O8qT)

File: share/goodie/cheat_sheets/json/bitwise-ops.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
+{
+    "id": "bitwise_ops_cheat_sheet",
+    "name": "Bitwise Operations",
+    "description": "A bitwise operator operates on one or more bit patterns or binary numerals at the level of their individual bits. It is a fast, simple action directly supported by the processor, and is used to manipulate values for comparisons and calculations.",
+
+    "metadata": {
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/bitwise-ops.json]: source name and source URL should be this [Bitwise operators](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Operators/Bitwise_Operators).

Good examples and description that's why!

File: share/goodie/cheat_sheets/json/bitwise-ops.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
+{
+    "id": "bitwise_ops_cheat_sheet",
+    "name": "Bitwise Operations",
+    "description": "A bitwise operator operates on one or more bit patterns or binary numerals at the level of their individual bits. It is a fast, simple action directly supported by the processor, and is used to manipulate values for comparisons and calculations.",
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/bitwise-ops.json]: make your description more precise! :+1:

File: share/goodie/cheat_sheets/json/bitwise-ops.json (hunk 1 above)
Comment (Sayan98) [inline share/goodie/cheat_sheets/json/bitwise-ops.json]: Alright. On it.

File: share/goodie/cheat_sheets/json/bitwise-ops.json (hunk 3 above)
Comment (Sayan98) [inline share/goodie/cheat_sheets/json/bitwise-ops.json]: As in lesser words?

Comment (amarlearning) [inline share/goodie/cheat_sheets/json/bitwise-ops.json]: @Sayan98 Description should be about your `cheat sheet` not about the `topic of cheatsheet`

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=bitwise ops cheatsheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/bitwise_ops_cheat_sheet_mobile.png?mPTY3WwI3v)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=bitwise ops cheatsheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/bitwise_ops_cheat_sheet_mobile.png?ggTIRSQxf4)

**Comment by Sayan98** [comment]
Hi @amarlearning don't you think the `<<<` is a bit restricted as I only see JS using it. Correct me if I am wrong.

**Comment by amarlearning** [comment]
@Sayan98 Are you making this cheat sheet for any specific language ? If `not` then there is no issue in including `Zero-fill right shift`!

**I think this is a general cheat sheet of Bitwise Operations, so it's ok to include `Zero-fill right shift`.**
_For less confusion, you can mention there that `Works only with JS`! :+1:_

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=bitwise ops cheatsheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/bitwise_ops_cheat_sheet_mobile.png?fGz47cC7DE)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=bitwise ops cheatsheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/bitwise_ops_cheat_sheet_mobile.png?F2t1G6XLRQ)

**Comment by Sayan98** [comment]
@amarlearning Any idea as to why the build is failing? All checks regarding this cheatsheet have passed. There's a problem regarding a different perl module as far as I can infer from the CI build log.

**Comment by gautamkrishnar** [comment]
@Sayan98 Failing build is not because of your code. 
cc: #3766

**Comment by gautamkrishnar** [comment]
This one is not specific to any languages right? @Sayan98

**Comment by Sayan98** [comment]
@gautamkrishnar Yes it's a generic cheatsheet, not language specific.

**Comment by Sayan98** [comment]
Hey @gautamkrishnar are we ready to merge?

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=bitwise ops cheatsheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/bitwise_ops_cheat_sheet_mobile.png?l1jMK8Exv8)

**Comment by daxtheduck** [comment]
@Sayan98 this is now merged! It should be live on DuckDuckGo within one week (or during the next release) and we'll be sure to notify you when it happens.

After your Instant Answer goes live for the World to use, you will be able to see [traffic information](http://docs.duckduckhack.com/maintaining/tools.html#monitor-your-instant-answer-page) about your IA as well as receive feedback directly from users. 
                
Please refer to the [maintenance guide and docs](http://docs.duckduckhack.com/maintaining/guidelines.html) to keep up with potential changes to your Instant Answer.
                
Thank you again for your help!

**Pull request merged**